manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+            if (assetId.toString() === NATIVE_TOKEN_ASSET_ID) {
+                const nativeBalance: any = await this.api.query.system.account(address);
+                return nativeBalance.data.free.toHuman();
+            } else {
+                const assetBalance: any = await this.api.query.assets.account(assetId, address);
+                if (assetBalance.value.isEmpty) {
+                    return "0";
+                } else {
+                    return assetBalance.value.balance.toString();
+                }
+            }
+        } catch (e) {
+            console.log("Failed to fetch public balance.");
+            console.error(e);
+        }
+    }
+
+    /// Executes a "To Private" transaction for any fungible token.
+    async toPrivateSend(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.toPrivateBuild(assetId,amount,polkadotSigner, polkadotAddress);
+        await this.sendTransaction(polkadotAddress,signed);
+        this.conditionalLog("To Private transaction finished.");
+    }
+
+    /// Builds and signs a "To Private" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async toPrivateBuild(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        try {
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            await this.setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await this.toPrivateFungible(assetId, amount);
+            const signResult = await this.signTransaction(null, transaction, this.network);
+            this.walletIsBusy = false;
+            return signResult;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Failed to build transaction.",e);
+        }
+    }
+
+    /// Executes a "Private Transfer" transaction for any fungible token.
+    async privateTransferSend(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.privateTransferBuild(assetId,amount,toPrivateAddress,polkadotSigner,polkadotAddress);
+        await this.sendTransaction(polkadotAddress,signed);
+        this.conditionalLog("Private Transfer transaction finished.");
+    }
+
+    /// Builds a "Private Transfer" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async privateTransferBuild(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        try {
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            await this.setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await this.privateFungibleTransfer(assetId, amount, toPrivateAddress);
```

Comment:
I would call `privateFungibleTransfer` -> `privateTransferBuildUnsigned`, I think the current naming is a bit confusing. Similarly I would call `toPrivateFungible` -> `toPrivateBuildUnsigned`, and `toPublicFungible` -> `toPublicBuildUnsigned`